frontend/src/service-worker.js

```diff
@@ -1,15 +1,54 @@
-import {serviceWorkerFetchListener} from "sync-message";
+/* eslint-disable */
+// Otherwise webpack can fail silently
+// https://github.com/facebook/create-react-app/issues/8014
+// Reference for service-worker.js: https://developers.google.com/web/tools/workbox/modules
 
-console.log(self.__WB_MANIFEST);
+import {serviceWorkerFetchListener} from 'sync-message'; // Do not remove
+const fetchListener = serviceWorkerFetchListener(); // Do not remove
+addEventListener('fetch', fetchListener); // Do not remove
 
-const fetchListener = serviceWorkerFetchListener();
+import {clientsClaim, skipWaiting} from 'workbox-core';
+import {ExpirationPlugin} from 'workbox-expiration';
+import {precacheAndRoute, createHandlerBoundToURL} from 'workbox-precaching';
+import {registerRoute} from 'workbox-routing';
+import {StaleWhileRevalidate} from 'workbox-strategies';
+import {CacheableResponsePlugin} from 'workbox-cacheable-response';
 
-addEventListener('fetch', fetchListener);
 
-addEventListener('install', function (e) {
-  e.waitUntil(self.skipWaiting());
-});
+clientsClaim();
 
-addEventListener('activate', function (e) {
-  e.waitUntil(self.clients.claim());
-});
+if (process.env.NODE_ENV === 'production') {
```

Comment:
I don't think this actually helps. The dev server doesn't serve service-worker.js, right? So we always essentially `npm run build` to use the service worker, which implies `NODE_ENV === 'production'` automatically. Even if service-worker.js is copied into `public` and the dev server is used, service-worker.js won't know that.

Reply:
I think we need a `REACT_APP_` env var so that we can control it, and turning it on would enable caching futurecoder-related things (i.e. everything other than pyodide) whether in prod or local dev. It would be nice if pyodide was always cached to speed up local dev, although I don't know if it actually makes a difference. For everything else, we don't want caching to interfere with normal non-service-worker development, so it should be off by default.

On that note, now that we're removing cross-origin isolation, the service worker is no longer optional and including it in development (without caching) needs to be more convenient. The instructions in https://github.com/alexmojaki/futurecoder#running-locally need to be updated to reflect that, and a command/script needs to be added to build and copy the service worker for the sake of `npm start`.

Reply:
As for:

> On that note, now that we're removing cross-origin isolation, the service worker is no longer optional and including it in development (without caching) needs to be more convenient. The instructions in [alexmojaki/futurecoder#running-locally](https://github.com/alexmojaki/futurecoder#running-locally) need to be updated to reflect that, and a command/script needs to be added to build and copy the service worker for the sake of npm start.

Draft of this README change: [5809506](https://github.com/alexmojaki/futurecoder/pull/320/commits/5809506d03d7ab169e32f88eb82647bbbab82839). But I'm unclear on this so I reverted it too ([de2b4cf](https://github.com/alexmojaki/futurecoder/pull/320/commits/de2b4cfd255fa094c0c25779ca7807ad8d280b1a)).  We can re-apply the change if it's actually what you meant.

Reply:
I am going to leave the README update to you because I'm just guessing at what you have in mind. It seems like you have a clearer sense of what you want the README update to look like.